src/hotspot/share/gc/shenandoah/shenandoahRegulatorThread.cpp

```diff
@@ -168,21 +175,14 @@ bool ShenandoahRegulatorThread::request_concurrent_gc(ShenandoahGenerationType g
   return accepted;
 }
 
-bool ShenandoahRegulatorThread::start_young_cycle() {
-  return _young_heuristics->should_start_gc() && request_concurrent_gc(YOUNG);
-}
-
-bool ShenandoahRegulatorThread::start_global_cycle() {
-  return _global_heuristics->should_start_gc() && request_concurrent_gc(ShenandoahControlThread::select_global_generation());
-}
-
 void ShenandoahRegulatorThread::stop_service() {
   log_info(gc)("%s: Stop requested.", name());
 }
 
 bool ShenandoahRegulatorThread::should_unload_classes() {
```

Comment:
I'd rename this to:

`should_start_metaspace_gc()`

to avoid confusion with similarly named method in other classes (heuristics) with a different semantics.

Relatedly, do we still need the second check in this method?

```
bool ShenandoahHeuristics::should_unload_classes() {
  if (!can_unload_classes()) return false;
  if (has_metaspace_oom()) return true;
  return ClassUnloadingWithConcurrentMark;
}
```

If not, it can be simplified to `return can_unload_classes() && ClassUnloadingWithConcurrentMark;`

I do believe, we should write a line documenting the intended semantics of `should_unload_classes()` in all these places because there ay otherwise be some confusion about what the truth modality "should" really means.

Reply:
I'll add comments to `should_unload_classes`, but I don't want to change the implementation. This method is used by other Shenandoah modes. As I think on it, I'm going to make the regulator honor any class unloading restrictions before starting a fruitless cycle that won't unload anything.